File: backend/app/models/article.py
from pydantic import BaseModel
from typing import Optional, List

class Article(BaseModel):
    id: str
    title: str
    summary: Optional[str] = None
    published: Optional[int] = None
    origin: Optional[str] = None
    url: Optional[str] = None
    categories: Optional[List[str]] = []
    related: Optional[List[str]] = []
    thumbnail_url: Optional[str] = None 

File: frontend/src/types/article.ts
export interface Article {
  id: string;
  title: string;
  summary: { content: string };
  url: string;
  published: number;
  origin: { title: string };
  categories: string[];
  related: { id: string; title: string; url: string }[];
  thumbnail_url?: string;
} 